pkg/sql/backfill/backfill.go

```diff
@@ -357,6 +358,15 @@ func (ib *IndexBackfiller) Init(
 		ib.types[i] = cols[i].Type
 	}
 
+	// Hydrate types used by the backfiller.
+	// TODO (rohany): As part of #49261, this needs to use cached enum data.
+	if err := ib.evalCtx.DB.Txn(evalCtx.Context, func(_ context.Context, txn *kv.Txn) error {
```

Comment:
This feels wrong to me. I think that the backfill protobuf should receive a fully hydrated type, don't you? I just don't quite understand why we don't send hydrated enum types over the wire in our distsql specs (or backfill specs). Right now, we'll be using a different txn than the one that started the backfill, which smells like trouble.

Reply:
I think it's becoming more apparent that I should at least try it and see how it looks. However, I don't think using a different txn is the problem, backfills use their own transactions throughout the process.

Reply:
But the transactions are allowed because of the guarantees about how the backfill works - they're on table data, which we know can only change in constrained ways. Getting type information during a backfill is racy, if someone were to update the enum in between starting the backfill and this txn, I feel like weirdness would ensue.

Reply:
I see. So passing the info to the backfiller works nicely for the index case, but I think we run into a similar problem when doing column backfills. If the column being backfilled has some expression that uses a user defined type, we'll need to access it (maybe that involves packaging up that information too), so we can't sidestep it completely. I'm not too worried about someone else updating the type, because the set of information in a type only "grows". I.e. people can only add things to new types, not remove them, so it shouldn't affect what version of the type we read.